api/v1alpha1/scheduledscaling_types.go

```diff
@@ -0,0 +1,166 @@
+	// TortoisePhasePartlyWorking means tortoise has maxReplicas and minReplicas recommendations ready,
+	// and applying the recommendation values.
+	// But, some of the resources are not scaled due to some reasons. (probably still gathering data)
+	TortoisePhasePartlyWorking TortoisePhase = "PartlyWorking"
+	// TortoisePhaseEmergency means tortoise is in the emergency mode.
+	TortoisePhaseEmergency TortoisePhase = "Emergency"
+	// TortoisePhaseBackToNormal means tortoise was in the emergency mode, and now it's coming back to the normal operation.
+	// During TortoisePhaseBackToNormal, the number of replicas of workloads are gradually reduced to the usual value.
+	TortoisePhaseBackToNormal TortoisePhase = "BackToNormal"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// ScheduledScaling is the Schema for the scheduledscalings API
+type ScheduledScaling struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ScheduledScalingSpec   `json:"spec,omitempty"`
+	Status ScheduledScalingStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// ScheduledScalingList contains a list of ScheduledScaling
+type ScheduledScalingList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ScheduledScaling `json:"items"`
+}
+
+type TargetRefs struct {
+	// ScaleTargetRef is the target of scaling.
+	// It should be the same as the target of HPA.
+	ScaleTargetRef CrossVersionObjectReference `json:"scaleTargetRef" protobuf:"bytes,1,name=scaleTargetRef"`
+	//Tortoise to be targeted for scheduled scaling
+	TortoiseName *string `json:"tortoiseName,omitempty" protobuf:"bytes,2,name=tortoiseName"`
+}
+
+// CrossVersionObjectReference contains enough information toet identify the referred resource.
+type CrossVersionObjectReference struct {
+	// kind is the kind of the referent; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	Kind string `json:"kind" protobuf:"bytes,1,opt,name=kind"`
+
+	// name is the name of the referent; More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+	Name string `json:"name" protobuf:"bytes,2,opt,name=name"`
+
+	// apiVersion is the API version of the referent
+	// +optional
+	APIVersion string `json:"apiVersion,omitempty" protobuf:"bytes,3,opt,name=apiVersion"`
+}
+
+type Schedule struct {
+	/// The schedule in Cron format, see https://en.wikipedia.org/wiki/Cron.
+	//start of schedule
+	StartAt string `json:"startAt" protobuf:"bytes,1,name=startAt"`
+	//end of schedule
+	FinishAt string `json:"finishAt" protobuf:"bytes,2,name=finishAt"`
+}
+
+type Strategy struct {
+	//Resource scaling requirements
+	Static Static `json:"static" protobuf: "bytes,1,name=static"`
+}
+
+type Static struct {
+	//Min replicas to be deployed on schedule
+	MinReplicas int `json:"minReplica" protobuf:"bytes,1,name=minReplica"`
```

Comment:
Sorry, please change it to `minimumMinReplicas`. 
https://github.com/mercari/tortoise/pull/378

We should allow Tortoise to set a higher replica number if it's recommended.